### ISSUE TITLE
When GHE root api page requires authentication Renovate fails to run

### ISSUE THREAD

**Issue by taha51595** [renovatebot/renovate#12479]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

28.23.1

### Describe the bug

When starting GHE will send a request to the configured endpoint (`https://hostname/api/v3/`)
https://github.com/renovatebot/renovate/blob/aa73a7bb22bd0945fc25b056ce4fa035bc44533f/lib/platform/github/index.ts#L105

The token in config is not included in this request, so if the endpoint requires authorization the request will fail and Renovate will stop. I'm not sure what the  `throwHttpErrors` option does, but the `head` method reuses the the `request` method of GithubHTTP which sets that option back to true - 
https://github.com/renovatebot/renovate/blob/6be7e756fcf699bb29d59d437ddecd3a24d32e23/lib/util/http/github.ts#L186

Including the `token` in the options fixed the issue locally.
```js
const gheQueryRes = await githubApi.head('/', { throwHttpErrors: false, token }); 
```

### Relevant debug logs

<details><summary>Logs</summary>

Call stack for request that is missing authorization header (modified code to log this)
```
Error
    at GithubHttp.request (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/util/http/index.js:141:21)
    at GithubHttp.request (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/util/http/github.js:188:28)
    at GithubHttp.head (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/util/http/index.js:213:17)
    at Proxy.initPlatform (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/platform/github/index.js:123:41)
    at initPlatform (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/platform/index.js:64:49)
    at globalInitialize (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/workers/global/initialize.js:62:48)
    at Object.start (/Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/workers/global/index.js:140:54)
    at async /Users/username/.nvm/versions/node/v14.18.0/lib/node_modules/renovate/dist/renovate.js:33:24
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment]
@rarkins @zharinov  Bad `GithubHttp` 😏 . Why is it always overriding `throwHttpErrors`?

**Comment by rarkins** [comment]
Don't know! Let's supply the token though as that's safer. I'll PR

**Comment by rarkins** [comment]
Incorporated into #12473

**Comment by rarkins** [comment]
Should be resolved in 28.23.2
